x-pack/solutions/security/plugins/security_solution/public/plugin.tsx

```diff
@@ -202,6 +204,21 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
       },
     });
 
+    productFeatureKeys$
+      .pipe(withLatestFrom(plugins.licensing.license$))
+      .subscribe(([productFeatureKeys, license]) => {
+        const isAssistantAvailable =
+          productFeatureKeys?.has(ProductFeatureAssistantKey.assistant) &&
+          license?.hasAtLeast('enterprise');
```

Comment:
I just realized that if `productFeatureKeys` is still _null_, or the license is not defined yet, this code will behave as if the assistant is not enabled, disabling the assistant management.
Can we add a check for at the beginning?

```suggestion
        if (!productFeatureKeys || !license) {
          return;
        }
        const isAssistantAvailable =
          productFeatureKeys.has(ProductFeatureAssistantKey.assistant) &&
          license.hasAtLeast('enterprise');
```

Reply:
It seems that as long as we use BehaviorSubject with an initial value, it will always emit the initial value as the first emission. To avoid that, I'd like to replace BehaviorSubject with Subject so the first emission will be the productFeatureKeys from ESS/serverless.
combineLatestWith waits for both productFeatureKeys$ and license$ to emit at least once before emitting a combined value.
In this case, we should be able to always receive the non null values.

Reply:
Using Subject the subscribe handler will be called only if it is registered before the first next call. If the next happens before the handler is subscribed, it will never be called. 
To avoid that, I'm sticking to BehaviorSubject and check if productFeatureKeys and license exists.